charts/nr-k8s-otel-collector/templates/service.yaml

```diff
@@ -0,0 +1,22 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: nr-k8s-otel-collector-gateway
```

Comment:
Do we really need a gateway? I fear that it could introduce a bottleneck and a component consuming 10x the memory and cpu.

Which is the down side against having each replica sending data autonomously?

Reply:
We're still considering using the gateway, unfortunately, we haven't had time yet to do load testing and cut down on ingest, the bottleneck is the big concern, but the main reason we want to have data written through the same collector is that there are some transforms we have to perform across multiple metrics that are collected separately via the agent pods (daemonset) and the gateway pod (deployment). Because otel does not yet fully support the kind of transforms we need to do, we are currently just not doing those transforms yet, but in the case that what we want to do becomes possible, we'll at least have to have _some_(not all, since bottleneck concern) metrics written through to the gateway from the daemonset. that being said, this is all hypothetical, because while have found a way to make those cross metric transforms in the NR platform, but it requires some engineering effort and isn't as simple as running a nrql query.  given this, we're going to keep the gateway service in here for now, as we may have to work on this sooner rather than later.